src/types.cpp

```diff
@@ -567,6 +567,9 @@ bool SizedType::FitsInto(const SizedType &t) const
   if (IsStringTy() && t.IsStringTy())
     return GetSize() <= t.GetSize();
 
+  if (IsIntegerTy() && t.IsIntegerTy() && (IsSigned() == t.IsSigned()))
+    return GetSize() <= t.GetSize();
```

Comment:
Nit: the comment for this method is outdated (it already was before this PR), should be adjusted or removed (I think that the code is quite self-documenting in this case).